### PR TITLE
Update rollback BBE

### PR DESCRIPTION
**Pull request by rdulmina**
## Purpose
$title

### CODE REVIEW

File: examples/rollback/rollback.bal [hunk 1]
```diff
@@ -47,3 +49,20 @@ function doUpdate(Update u) returns error? {
 
     return;
 }
+
+function incrementSallary(int[] salaryList) returns error? {
```
Comment (pcnfernando) [inline examples/rollback/rollback.bal:53]:
```suggestion
function incrementSalary(int[] salaryList) returns error? {
```

File: examples/rollback/rollback.bal [hunk 2]
```diff
@@ -15,11 +16,13 @@ public function main() returns error? {
...
-    // the error is returned from the `main` and it exits.
-    check transfer(updates);
 
-    return;
+    io:println(transfer(updates));
+
+    // Creates an array of employee salaries.
+    int[] salaryList = [100, 200, 300, 100];
+
+    incrementSallary(salaryList);
```
Comment (pcnfernando) [inline examples/rollback/rollback.bal:25]: error assignment required

File: examples/rollback/rollback.bal [hunk 3]
```diff
@@ -15,11 +16,13 @@ public function main() returns error? {
...
     ];
-    // If an error is returned from the `transfer` function,
-    // the error is returned from the `main` and it exits.
-    check transfer(updates);
 
-    return;
+    io:println(transfer(updates));
+
+    // Creates an array of employee salaries.
+    int[] salaryList = [100, 200, 300, 100];
```
Comment (pcnfernando) [inline examples/rollback/rollback.bal:23]: Shall we maybe use the `updates` list or some other list with the same context?

File: examples/rollback/rollback.bal [hunk 4]
```diff
@@ -14,11 +15,19 @@ public function main() returns error? {
         {updateIndex: 3, stockMnt: -1000},
         {updateIndex: 4, stockMnt: -2000}
     ];
-    // If an error is returned from the `transfer` function,
-    // the error is returned from the `main` and it exits.
-    check transfer(updates);
 
-    return;
+    io:println(transfer(updates));
```
Comment (lasinicl) [inline examples/rollback/rollback.bal:19]: Shall we add a comment for this, that the transaction is implicitly rolled back due to the error?

File: examples/rollback/rollback.bal [hunk 5]
```diff
@@ -14,11 +15,19 @@ public function main() returns error? {
...
+
+    // Creates an array of employee salaries.
+    int[] salaryList = [100, 200, 300, 100];
+
+    // This salary increment will rollback
```
Comment (lasinicl) [inline examples/rollback/rollback.bal:24]:
```suggestion
    // This salary increment will be rolled back.
```

File: examples/rollback/rollback.bal [hunk 6]
```diff
@@ -14,11 +15,19 @@ public function main() returns error? {
...
+    check incrementSalary(salaryList);
+
+    int[] salaryList2 = [100, 200, 100, 100];
+
+    // This salary increment will be successful
```
Comment (lasinicl) [inline examples/rollback/rollback.bal:29]:
```suggestion
    // This salary increment will be successful.
```

File: examples/rollback/rollback.bal [hunk 7]
```diff
@@ -44,3 +53,21 @@ function doUpdate(Update u) returns error? {
...
     return;
 }
+
+function incrementSalary(int[] salaryList) returns error? {
+    transaction {
+        foreach int index in 0 ..< salaryList.length() {
+            salaryList[index] += 100;
+        }
+
+        // If the new total salary exceeds `1000`, then the rollback statement performs
```
Comment (lasinicl) [inline examples/rollback/rollback.bal:63]: Do we need `1000` within backticks? :thinking: `doUpdate()` function mentions `-1500` without the backticks.  Maybe we can update both to be consistent with whatever the preferred way.

File: examples/rollback/rollback.out [hunk 8]
```diff
@@ -1,2 +1,4 @@
-$ bal run rollback.bal
-error: Not enough Stocks:  {"stockIndex":4}
+bal run rollback.bal
```
Comment (lasinicl) [inline examples/rollback/rollback.out:1]:
```suggestion
$ bal run rollback.bal
```

File: examples/rollback/rollback.bal (hunk 4 above)
Comment (rdulmina) [reply examples/rollback/rollback.bal:19]: That is explained inside `transfer` function

Comment (rdulmina) [reply examples/rollback/rollback.bal:19]: Will also add a comment there as well

**Review by lasinicl — APPROVED** [review]

File: examples/rollback/rollback.bal [hunk 9]
```diff
@@ -37,10 +46,28 @@ function transfer(Update[] updates) returns error? {
...
```
Comment (praneesha) [inline examples/rollback/rollback.bal:63]:
```suggestion
        // If the new total salary exceeds `1000`, then, the rollback statement performs a
```

File: examples/rollback/rollback.out [hunk 10]
```diff
@@ -1,2 +1,4 @@
 $ bal run rollback.bal
-error: Not enough Stocks:  {"stockIndex":4}
+error("Not enough Stocks: ",stockIndex=4)
```
Comment (praneesha) [inline examples/rollback/rollback.out:2]:
```suggestion
error("Not enough stocks: ",stockIndex=4)
```

File: examples/rollback/rollback.bal [hunk 11]
```diff
@@ -37,10 +46,28 @@ function transfer(Update[] updates) returns error? {
...
+
+function incrementSalary(int[] salaryList) returns error? {
+    transaction {
+        foreach int index in 0 ..< salaryList.length() {
+            salaryList[index] += 100;
+        }
+
+        // If the new total salary exceeds `1000`, then, the rollback statement performs 
+        // rollback on the transaction.
+        if (salaryList.reduce(function(int total, int n) returns int => total + n, 0) > 1000) {
```
Comment (pcnfernando) [inline examples/rollback/rollback.bal:65]: Shall we use a simpler version
```suggesion
        int salarySum = int:sum(...salaryList);
        if (salarySum > 1000) {
```

File: examples/rollback/rollback.bal [hunk 12]
```diff
@@ -37,10 +46,28 @@ function transfer(Update[] updates) returns error? {
...
+            io:println("Salary limit exceeded");
```
Comment (pcnfernando) [inline examples/rollback/rollback.bal:66]:
```suggestion
            io:println("Budget exceeded");
```

**Pull request merged**